junit5-engine/src/main/java/org/junit/gen5/engine/junit5/descriptor/MethodTestDescriptor.java

```diff
@@ -57,6 +54,17 @@
 	}
 
 	@Override
+	public String getUniqueId() {
+		// @formatter:off
+		return String.format("%s:%s#%s(%s)",
+				getTestEngine().getId(),
+				getTestClass().getName(),
+				getTestMethod().getName(),
+				ObjectUtils.nullSafeToString(getTestMethod().getParameterTypes()));
+		// @formatter:on
+	}
```

Comment:
Yep, it is correct (we have tests for this). Anyways, I am with you, we should investigate some more time on this. I would like to have a separate ticket for it, as it is not as easy as it looks in the first place. If you like, we could pair on this topic and have a look at your "path" idea.

I have considered to create the unique id using the parent, but in my first version this did not work out, as I moved methods from the specification on the top level. After we decided to create the whole tree for every node, it is now possible to use the parent for id generation. As I said, I would like to consider this topic in an own issue, as I think it needs some more time to find out all dependencies.